### PR TITLE
Display latest version in header

### PR DESCRIPTION
**Pull request by jekyllbot**
PR automatically created for @DirtyF.

```text
Display latest version in header
```

### CODE REVIEW

**Comment by DirtyF** [comment]
I don't like to have to update the config file manually at each release, any way to add this to our `rake release` task?

**Comment by ashmaroli** [comment]
https://github.com/jekyll/jekyll/blob/5ef2debefba10eb8f4295f79fa4bee9dcf2961fb/rake/site.rake#L87-L90

**Comment by DirtyF** [comment]
@ashmaroli I was thinking of a task that parses `docs/_config.yml` and replace the `version` value. 😄

**Comment by ashmaroli** [comment]
yes, that's the way to do it.. I just pointed you to where to add the code..
Do you want me to push to this branch or just paste the code here..?

**Comment by DirtyF** [comment]
@ashmaroli as you like (I was goint to try to learn how to parse a yml file in Ruby, but I 'm sure you'll be quicker at this). Thanks for your help.

**Comment by ashmaroli** [comment]
This is the updated block for above snippet.. I'll leave it to you to update the **task name(s) and description** and test locally .. :wink: :grin:
```ruby
  desc "Write the site latest_version.txt file"
  task :version_file do
    return if version =~ %r!(beta|rc|alpha)!i
    config_file = File.join(docs_folder, "_config.yml")
    contents = File.read(config_file)
    File.write(config_file, contents.sub(/(?:version\s*:\s+)(.+)/, "version: #{version}"))
  end
```

**Comment by DirtyF** [comment]
@ashmaroli thanks, works like a charm 👍

File: rake/site.rake [hunk 1]
```diff
@@ -84,9 +84,12 @@ namespace :site do
     siteify_file(".github/SUPPORT.markdown", "title" => "Support")
   end
 
-  desc "Write the site latest_version.txt file"
-  task :version_file do
-    File.open("#{docs_folder}/latest_version.txt", "wb") { |f| f.puts(version) } unless version =~ %r!(beta|rc|alpha)!i
+  desc "Write the site latest version"
```
Comment (ashmaroli) [inline rake/site.rake]: How about..?
```diff
- "Write the site latest version"
+ Write the latest Jekyll version used to generate the site"
```

Comment (DirtyF) [reply rake/site.rake]: I kept the beginning of your suggestion, site is generated by GitHub Pages and as now, it may not use latest version.

**Comment by ashmaroli** [comment]
okay.. I'll let the others review the changes..
If you (and the others) feel like taking the task to another level, the following patch should insert the version data if its ever removed..
```ruby

  desc "Write the latest Jekyll version"
  task :latest_version do
    return if version =~ %r!(beta|rc|alpha)!i
    config_file   = File.join(docs_folder, "_config.yml")
    contents      = File.read(config_file)
    version_regex = /(?:version\s*:\s+)(.+)/

    if contents =~ version_regex
      File.write(config_file, contents.sub(version_regex, "version: #{version}"))
    else
      File.open(config_file, "a+") do |file|
        file.puts <<-TXT

# latest docs version displayed in the header
# inserted and updated automatically by the following rake task(s)
#   bundle exec rake site:preview
#   bundle exec rake site:latest_version
version: #{version}"
TXT
      end
    end
  end
```

**Comment by DirtyF** [comment]
@jekyllbot: merge +site

**Comment by Crunch09** [comment]
@DirtyF @ashmaroli  ❤️ thanks for fixing this so quickly!
A nicer alternative might be:
```ruby
require 'safe_yaml/load'
config = SafeYAML.load_file(config_file)
config["version"] = version
File.write(config_file, YAML.dump(config))
```
so we don't have to use a regex, but just a suggestiong as it is already merged and works fine!

**Comment by DirtyF** [comment]
@Crunch09 👍  that's exactly what I was thinking about, I like it DRY. 

Feel free to open a PR and I'll merge it 😄

**Comment by ashmaroli** [comment]
> A nicer alternative might be:

Wow @Crunch09, That is sooo much better..!! ( Why didn't I think of that!! :smile: )

**Pull request merged**